### ISSUE TITLE
Possible temporary failure edge case with no automatic recovery

### ISSUE THREAD

**Issue by Marcuzz** [vippsas/vipps-recurring-woocommerce#19]
I noticed an order that had the following postmeta:

```
_vipps_recurring_captured: 1
```

and 

```
_vipps_recurring_pending_charge: 1
_vipps_recurring_locked_for_update: 1
```

This means the charge was captured, but it failed before it was able to set pending_charge to 0 and update the order status. Perhaps a temporary failure with a certain API request?

This put the order in a state where it was not automatically recoverable and manual intervention was needed.

**Comment by Marcuzz** [comment]
Solution: Make a time based lock instead

**Comment by Marcuzz** [comment]
Fixed in 1.3.10
